### ISSUE TITLE
Several keys with identical key codes in AZERTY

### ISSUE THREAD

**Issue by Haroenv** [wesbos/keycodes#88]
For example the 

* <kbd>shift</kbd><kbd>,</kbd> (?) and <kbd>shift</kbd><kbd>:</kbd> (/) both give keycode `191`. 
* The <kbd>=</kbd> and <kbd>shift</kbd><kbd>=</kbd> (+) both give `187`
* The numbers (for which you normally need to press <kbd>shift</kbd>) all show as their keycode without shift, except <kbd>4</kbd>, which shows <kbd>'</kbd> `222` and <kbd>4</kbd> `52`when <kbd>shift</kbd> is also pressed
* the @# key shows in both cases `192`, a key which on Windows keyboards shows superscript 2 and 3

reference image

![](http://forums.newtek.com/attachment.php?attachmentid=62455&d=1219945745)

Not really sure how this can be resolved since there will be loads of differences per language and different keys with the same keycode.

**Comment by bantic** [comment]
This project only reports the `keyCode` of the key that the user last hit, so I think it makes sense that it displays the same keyCode for <kbd>shift</kbd><kbd>=</kbd> and <kbd>=</kbd>. In both cases the last keydown was for the <kbd>=</kbd> key, even though in one case there is a previous key that is still being held down. For example, the keystrokes <kbd>command</kbd><kbd>=</kbd>, <kbd>control</kbd><kbd>=</kbd> and <kbd>option</kbd><kbd>=</kbd> all also print the keyCode for the <kbd>=</kbd> key, since in each case that's the key that the user hit last. The *character* that would be printed for that key is different if <kbd>shift</kbd> is also depressed (and to determine that you'd look at the event's `charCode` or similar property), but it's always the same physical key on the keyboard.

**Comment by Haroenv** [comment]
Makes sense, but for other keys they do have separate keycodes; for example <kbd>,</kbd>  and <kbd>shift</kbd><kbd>,<kbd> (`?`) are respectively `188` and `191`. I suspect this is more a browser/spec problem than a problem with this implementation though

**Comment by bantic** [comment]
> respectively `188` and `191`

Yeah, that is a head-scratcher (!). And I think you're right that it's browser/spec-related.

**Comment by ItamarShDev** [comment]
> Makes sense, but for other keys they do have separate keycodes; for example , and shift , (`?`) are respectively `188` and `191`. I suspect this is more a browser/spec problem than a problem with this implementation though

working on the product of the company I work for, I can tell you that the ASCII codes for AZERTY are messed up, 6 and + have the same mapping if I remember correctly.

**Comment by cwadrupldijjit** [comment]
From what it looks like on the [`KeyboardEvent` documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent), the reason for this and similar issues is the fact that the event.which and the event.code are both deprecated or not suggested for use, the reason for it being that they don't really support cleanly the current keyboard layout of the user.  I think generally using the `event.key` as well as the `event.location` is a better option for this.

I think this is less of a failing of this tool and more a failing of the browsers.

Additionally, I think that the real improvement would come from adding an `event.location` cell in the table on the bottom.  I may do that when I have some time, though I've noticed that there are some PRs in here from years ago with little to no explanation as to why they weren't merged.  Is this project still being maintained?
